File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/rendertaskusageview.rb
module Asposetasksjava
  module RenderTaskUsageView
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Define the SaveOptions with required TimeScale settings as Days
		options = Rjb::import('com.aspose.tasks.PdfSaveOptions').new
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').Days)
		# Set the Presentation format to ResourceUsage
		options.setPresentationFormat(Rjb::import('com.aspose.tasks.PresentationFormat').TaskUsage)
		# Save the Project
		project.save(data_dir + "task_days.pdf", options)

		# Set the Tiemscale settings to ThirdsOfMonths
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').ThirdsOfMonths)
		# Save the Project
		project.save(data_dir + "task_thirdsOfMonths.pdf", options)

		# Set the Timescale settings to Months
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').Months)
		# Save the project
		project.save(data_dir + "task_months.pdf", options)

		puts "Created task usage view files."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/RenderTaskUsageView.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\PdfSaveOptions as PdfSaveOptions;
use com\aspose\tasks\Timescale as Timescale;
use com\aspose\tasks\PresentationFormat as PresentationFormat;

class RenderTaskUsageView{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Define the SaveOptions with required TimeScale settings as Days
		$options = new PdfSaveOptions();
        $timescale=new Timescale();
        $options->setTimescale($timescale->Days);

		# Set the Presentation format to ResourceUsage
        $presentationFormat=new PresentationFormat();
		$options->setPresentationFormat($presentationFormat->TaskUsage);
		# Save the Project
		$project->save($dataDir . "task_days.pdf", $options);

		# Set the Tiemscale settings to ThirdsOfMonths
		$options->setTimescale($timescale->ThirdsOfMonths);
		# Save the Project
		$project->save($dataDir . "task_thirdsOfMonths.pdf", $options);

		# Set the Timescale settings to Months
		$options->setTimescale($timescale->Months);
		# Save the project
		$project->save($dataDir . "task_months.pdf", $options);

		print "Created task usage view files.".PHP_EOL;
    }

}

?>